packages/typespec-ts/src/modular/helpers/clientHelpers.ts

```diff
@@ -73,14 +73,15 @@ export function getClientParameters(
   const hasDefaultValue = (p: SdkParameter) =>
     p.clientDefaultValue || p.__raw?.defaultValue || p.type.kind === "constant";
   const isRequired = (p: SdkParameter) =>
-    !p.optional &&
-    ((!hasDefaultValue(p) &&
+    // Special case: when apiVersionAsRequired is true, apiVersion should always be considered required
```

Comment:
I don't think this fix is correct. We should trust always the defined optionality for api-version. If it is optional in spec, why we set it as required?